### PR TITLE
fix(resource-monitor): get DW Namespace name from ENV variable

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

### What does this PR do?
Reads namespace's name form the environment variable

depends on devworkspace-operator 0.12.0 https://github.com/devfile/devworkspace-operator/pull/695

### Screenshot/screencast of this PR
Screenshot from 2021-12-07 14-48-03

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20800

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)
- [ ] Optional Companion PR for updating the HappyPath tests is approved and ready to be merged

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: plugins/resource-monitor-plugin/src/plugin.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ export async function start(context: theia.PluginContext): Promise<void> {
 }
 
 export async function getNamespace(): Promise<string> {
+  if (process.env.DEVWORKSPACE_NAMESPACE !== undefined) {
```
Comment (benoitf) [inline plugins/resource-monitor-plugin/src/plugin.ts:30]: it looks like we're breaking abstraction of the che API and bypassing it using directly the underlying environment variables

getNamespace() should be a method of che.devfile or che.workspace API  and then implementation for che-server or devWorkspaces would be different

**Review by benoitf — APPROVED** [review]
approving as it's still better than what we have today

File: plugins/resource-monitor-plugin/src/plugin.ts (hunk 1 above)
Comment (svor) [reply plugins/resource-monitor-plugin/src/plugin.ts]: @benoitf  I've added `getCurrentNamespace` method into `che.workspace` API. Could you please take a look at the changes again.

**Comment by svor** [comment]
[crw-ci-test]

**Review by benoitf — APPROVED** [review]

File: plugins/resource-monitor-plugin/src/plugin.ts (hunk 1 above)
Comment (benoitf) [reply plugins/resource-monitor-plugin/src/plugin.ts]: thanks

**Pull request merged**